Helper/SiteAccess.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\RecommendationBundle\Helper;
+
+use eZ\Publish\Core\Base\Exceptions\NotFoundException;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\Core\MVC\Symfony\SiteAccess as CurrentSiteAccess;
+use LogicException;
+
+class SiteAccess
+{
+    const SYSTEM_DEFAULT_SITEACCESS_NAME = 'default';
+
+    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
+    private $configResolver;
+
+    /** @var \eZ\Publish\Core\MVC\Symfony\SiteAccess */
+    private $siteAccess;
+
+    /** @var array */
+    private $siteAccessConfig;
+
+    /** @var string */
+    private $defaultSiteAccessName;
+
+    public function __construct(
+        ConfigResolverInterface $configResolver,
+        CurrentSiteAccess $siteAccess,
+        $siteAccessConfig,
+        $defaultSiteAccessName = self::SYSTEM_DEFAULT_SITEACCESS_NAME
+    ) {
+        $this->configResolver = $configResolver;
+        $this->siteAccess = $siteAccess;
+        $this->siteAccessConfig = $siteAccessConfig;
+        $this->defaultSiteAccessName = $defaultSiteAccessName;
+    }
+
+    /**
+     * Returns rootLocation by siteAccess name or by default siteAccess.
+     *
+     * @param string|null $siteAccessName
+     *
+     * @return int
+     */
+    public function getRootLocationBySiteAccessName($siteAccessName = null)
+    {
+        return $this->configResolver->getParameter(
+            'content.tree_root.location_id',
+            null,
+            $siteAccessName ?: $this->siteAccess->name
+        );
+    }
+
+    /**
+     * Returns list of rootLocations from siteAccess list.
+     *
+     * @param array $siteAccesses
+     *
+     * @return array
+     */
+    public function getRootLocationsBySiteAccesses(array $siteAccesses)
+    {
+        $rootLocations = [];
+
+        foreach ($siteAccesses as $siteAccess) {
+            $rootLocationId = $this->getRootLocationBySiteAccessName($siteAccess);
+
+            $rootLocations[$rootLocationId] = $rootLocationId;
+        }
+
+        return array_keys($rootLocations);
+    }
+
+    /**
+     * Returns languages based on mandatorId or siteaccess.
+     *
+     * @param null|int $mandatorId
```

Comment:
`mandatorId` is the customer ID from eZ Recommendation.

Reply:
Ok, @damianz5 could you add such description to each `$mandatorId` argument used in the code? I asked because Oxford's English dictionary does know that word and so it's not very descriptive.